backend/endpoints/checksum/entrypoint.py

```diff
@@ -0,0 +1,61 @@
+from typing import BinaryIO, Dict
+from urllib.parse import urlparse
+
+import boto3
+from jsonschema import ValidationError, validate
+from multihash import FUNCS
+
+from ..utils import error_response, success_response
+
+REQUEST_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "httpMethod": {"type": "string", "enum": ["GET"]},
+        "body": {
+            "type": "object",
+            "properties": {
+                "href": {"type": "string", "pattern": "^s3://"},
+                "checksum:multihash": {"type": "string"},
+            },
+            "required": ["href", "checksum:multihash"],
+        },
+    },
+    "required": ["httpMethod", "body"],
+}
+
+
+def lambda_handler(event: Dict, context: Dict) -> Dict:
+    try:
+        validate(event, REQUEST_SCHEMA)
+    except ValidationError as error:
+        return error_response(400, error.message)
+
+    s3 = getattr(context, "s3", None)  # For testing
+    if s3 is None:
+        s3 = boto3.client("s3")
+
+    href = event["body"]["href"]
+    url = urlparse(href, allow_fragments=False)
+    try:
+        s3_object = s3.get_object(Bucket=url.netloc, Key=url.path)
+    except s3.exceptions.NoSuchKey:
+        return error_response(400, f"Could not read URL: '{href}'")
+
+    except s3.exceptions.ClientError as error:
+        return error_response(
+            400,
+            f"{error.response['Error']['Code']}: {error.response['Error']['Message']}",
+        )
+
+    if not verify_stream_multihash(s3_object["Body"], event["body"]["checksum:multihash"]):
+        return error_response(400, "Checksum mismatch. Is the file corrupted?")
+
+    return success_response(200, {})
+
+
+def verify_stream_multihash(object_stream: BinaryIO, hex_multihash: str) -> bool:
+    checksum_function_code = int(hex_multihash[:2], 16)
+    checksum_function = FUNCS[checksum_function_code]
+    file_contents = object_stream.read()
```

Comment:
Might we want to read this in chunk just in case the file is large?

Reply:
Oh, definitely. I wanted to get the [S3 teardown](https://stackoverflow.com/q/64961655/96588) to work before writing a test for that. I also want to test with different chunk sizes to make sure it's as fast as possible.

Reply:
I've changed the code to chunk reads and added issue #71 to optimize it once we're further along.